Assets/SEE/Controls/KeyBindings.cs

```diff
@@ -311,8 +312,29 @@ public static void SetBindingForKey(KeyActionDescriptor descriptor, KeyCode keyC
             }
             else
             {
+                KeyCode oldKey = descriptor.KeyCode;
                 descriptor.KeyCode = keyCode;
+                string path = Application.dataPath + "/StreamingAssets/KeyBindings.json";
```

Comment:
This code must become a method of class `KeyBindings`. See my comment below.

Moreover, the key binding is a mapping of keys onto actions. You are just writing the keys and assume their number stays the same and that there is always a fixed order. Neither should be asumed. What do you do if we later add or remove a key binding? Furthermore, the file in which the key bindings are stored must be understood and possibly edited by a human. Hence, the interpretation of its content must be simple and clear. The content should be a JSON structure of key-value pairs. The JSON key is the key code and the JSON value is the action.

Reply:
Well, the case that key bindings get removed or added, is handelt by keyBindingDataList.Count() == KeyBindings.GetKeyCodes().Count(), but yes if someone changes the order of the bindings, then I have a big problem. Im going to fix it.

Reply:
@koschke please check again, if its better now.

Reply:
Much better now.